app.py

```diff
@@ -181,8 +231,8 @@ def get_country_select_options(countries):
     Input("stored-basic-country-data", "data"),
 )
 def fetch_country_data_once(countries, subnational_data, country_data):
-    countries = [x["label"] for x in countries]
-    if country_data is None:
+    if country_data is None and current_user and current_user.is_authenticated:
```

Comment:
I'm not a fan of the repeated checks for each of the `fetch_*data_once` calls. @yukinko-iwasaki I'm going to merge this PR for now. Can you try refactoring the checks? Perhaps see if they can be moved to the QueryService or something?